pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -715,6 +716,33 @@ def _to_abstract_repr(self) -> dict[str, Any]:
         d["is_virtual"] = False
         return d
 
+    @staticmethod
+    def from_abstract_repr(obj_str: str) -> Device:
```

Comment:
On the cloud side can we simply use the `VirtualDevice.from_abstract_repr` method to build our device object? I am not super up to date on the differences between Devices and VirtualDevices. 

We build the device object to 1) validate the layout 2) switch the device of the sequence for validation.

In our current code I see that we only validate the layout if the device is not virtual, is it correct virtual devices do not have pre calibrated layouts?

Reply:
It seems that what you are describing is QPU validation, so what you want is `Device.from_abstract_repr()`.

A `VirtualDevice` is not tied to any physical implementation, which is why it doesn't have certain parameters like `pre_calibrated_layouts`. I'd say you probably won't have a need for `VirtualDevice.from_abstract_repr()` unless you are deserializing a device for execution on an emulator at some point.